cpp/include/cudf/text/detail/tokenize.hpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column.hpp>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+
+namespace cudf
+{
+namespace nvtext
```

Comment:
Can we rename the namespace to just `text`?

```suggestion
namespace text
```

Reply:
As mentioned in the #3965, porting these nvtext functions directly into libcudf is just a temporary measure. So I wanted to use 'nvtext' for the namespace to identify the functions/features we should move out later.

Reply:
it's not decided whether this is a temporary measure yet. Lots of things have been temporary measures in cuDF in the past and ended up becoming "permanent".  `namespace text` is just as easy to search for as `namespace nvtext`.

The more I think about it, tokenization of strings columns is suited to the existing `strings` namespace.

Reply:
The `strings` namespace has `split` function like the one for Pandas strings which is a kind of tokenize. The `tokenize` methods here are specialized more for the needs of common NLP operations.

Reply:
@randerzander @kkraus14

Reply:
Actually, I should not have included the `cudf` namespace but should just use `nvtext` or perhaps `cutext` for the base namespace here.